Shaka Typescript definition overriding Google Cast Sender with incomplete defs

**Have you read the [FAQ](https://bit.ly/ShakaFAQ) and checked for duplicate open issues?**
Yes, this does probably relate to #1030, but I assume it is cleaner to report it as a separate bug.

**What version of Shaka Player are you using?**
3.0.8

**Can you reproduce the issue with our latest release version?**
NA

**Can you reproduce the issue with the latest code from `master`?**
Not tested

**Are you using the demo app or your own custom app?**
Own

**If custom app, can you reproduce the issue using our demo app?**
NA

**What browser and OS are you using?**
Any

**For embedded devices (smart TVs, etc.), what model and firmware version are you using?**
NA

**What are the manifest and license server URIs?**
NA

**What did you do?**
In code unrelated to the Shaka based player Typescript gives errors, example:
```
const sessionObj = session.getSessionObj();
sessionObj.addMediaListener(media => {
  onMediaSession(media);
});
```

**What did you expect to happen?**
Type from `node_modules/@types/chrome/chrome-cast/index.d.ts` to be resolved. (That is included when using the `@types/chromecast-caf-sender` package.)

**What actually happened?**
Type from `node_modules/shaka-player/dist/shaka-player.compiled.d.ts` seems to override, in this case:
```
// Generated from /usr/local/google/home/joeyparrish/hacking/shaka/clean/externs/chromecast.js
declare namespace chrome.cast {
  class Session {
    ...
```
Which does not contain the `addMediaListener` method.

I see there are similar issues reported for IMA types in #1030, maybe Shaka should not export these definitions for 'externs'? or maybe not in the global scope at least?

```
node_modules/shaka-player/dist/shaka-player.compiled.d.ts:446:62 - error TS2694: Namespace 'google.ima.dai.api' has no exported member 'StreamRequest'.

446     requestServerSideStream (imaRequest : google.ima.dai.api.StreamRequest , backupUrl ? : string ) : Promise < string > ;
                                                                 ~~~~~~~~~~~~~

node_modules/shaka-player/dist/shaka-player.compiled.d.ts:574:37 - error TS2694: Namespace 'google.ima' has no exported member 'Ad'.

574     constructor (imaAd : google.ima.Ad , imaAdManager : google.ima.AdsManager ) ;
                                        ~~

node_modules/shaka-player/dist/shaka-player.compiled.d.ts:599:45 - error TS2694: Namespace 'google.ima.dai.api' has no exported member 'Ad'.

599     constructor (imaAd : google.ima.dai.api.Ad | null , video : HTMLMediaElement | null ) ;
                                                ~~

node_modules/shaka-player/dist/shaka-player.compiled.d.ts:2269:9 - error TS2300: Duplicate identifier 'ApiConfig'.

2269   class ApiConfig {
             ~~~~~~~~~

  node_modules/@types/chrome/chrome-cast/index.d.ts:206:18
    206     export class ApiConfig {
                         ~~~~~~~~~
    'ApiConfig' was also declared here.

node_modules/shaka-player/dist/shaka-player.compiled.d.ts:2276:9 - error TS2300: Duplicate identifier 'Error'.

2276   class Error {
             ~~~~~

  node_modules/@types/chrome/chrome-cast/index.d.ts:232:18
    232     export class Error {
                         ~~~~~
    'Error' was also declared here.

node_modules/shaka-player/dist/shaka-player.compiled.d.ts:2286:9 - error TS2300: Duplicate identifier 'Receiver'.

2286   class Receiver {
             ~~~~~~~~

  node_modules/@types/chrome/chrome-cast/index.d.ts:452:18
    452     export class Receiver {
                         ~~~~~~~~
    'Receiver' was also declared here.

node_modules/shaka-player/dist/shaka-player.compiled.d.ts:2293:9 - error TS2300: Duplicate identifier 'Session'.

2293   class Session {
             ~~~~~~~

  node_modules/@types/chrome/chrome-cast/index.d.ts:300:18
    300     export class Session {
                         ~~~~~~~
    'Session' was also declared here.

node_modules/shaka-player/dist/shaka-player.compiled.d.ts:2309:9 - error TS2300: Duplicate identifier 'SessionRequest'.

2309   class SessionRequest {
             ~~~~~~~~~~~~~~

  node_modules/@types/chrome/chrome-cast/index.d.ts:280:18
    280     export class SessionRequest {
                         ~~~~~~~~~~~~~~
    'SessionRequest' was also declared here.

node_modules/shaka-player/dist/shaka-player.compiled.d.ts:2316:7 - error TS2451: Cannot redeclare block-scoped variable 'STOPPED'.

2316   let STOPPED : string ;
           ~~~~~~~

  node_modules/@types/chrome/chrome-cast/index.d.ts:109:9
    109         STOPPED = "stopped"
                ~~~~~~~
    'STOPPED' was also declared here.

node_modules/shaka-player/dist/shaka-player.compiled.d.ts:2320:7 - error TS2451: Cannot redeclare block-scoped variable 'isAvailable'.

2320   let isAvailable : boolean ;
           ~~~~~~~~~~~

  node_modules/@types/chrome/chrome-cast/index.d.ts:122:16
    122     export var isAvailable: boolean;
                       ~~~~~~~~~~~
    'isAvailable' was also declared here.

node_modules/shaka-player/dist/shaka-player.compiled.d.ts:2364:41 - error TS2694: Namespace 'google.ima' has no exported member 'AdDisplayContainer'.

2364     constructor (container : google.ima.AdDisplayContainer ) ;
                                             ~~~~~~~~~~~~~~~~~~

node_modules/shaka-player/dist/shaka-player.compiled.d.ts:2368:34 - error TS2694: Namespace 'google.ima' has no exported member 'ImaSdkSettings'.

2368     getSettings ( ) : google.ima.ImaSdkSettings | null ;
                                      ~~~~~~~~~~~~~~

node_modules/shaka-player/dist/shaka-player.compiled.d.ts:2446:130 - error TS2694: Namespace 'google.ima.dai.api' has no exported member 'UiSettings'.

2446     constructor (videoElement : HTMLMediaElement | null , adUiElement ? : HTMLElement | null , uiSettings ? : google.ima.dai.api.UiSettings | null ) ;
                                                                                                                                      ~~~~~~~~~~

node_modules/shaka-player/dist/shaka-player.compiled.d.ts:2455:55 - error TS2694: Namespace 'google.ima.dai.api' has no exported member 'StreamRequest'.

2455     requestStream (streamRequest : google.ima.dai.api.StreamRequest | null ) : any ;
                                                           ~~~~~~~~~~~~~

node_modules/shaka-player/dist/shaka-player.compiled.d.ts:2463:29 - error TS2694: Namespace 'google.ima' has no exported member 'ImaSdkSettings'.

2463   let settings : google.ima.ImaSdkSettings ;
                                 ~~~~~~~~~~~~~~

node_modules/shaka-player/dist/shaka-player.compiled.d.ts:2799:62 - error TS2694: Namespace 'google.ima.dai.api' has no exported member 'StreamRequest'.

2799     requestServerSideStream (imaRequest : google.ima.dai.api.StreamRequest , backupUrl ? : string ) : Promise < string > ;
                                                                  ~~~~~~~~~~~~~

node_modules/@types/chrome/chrome-cast/index.d.ts:109:9 - error TS2451: Cannot redeclare block-scoped variable 'STOPPED'.

109         STOPPED = "stopped"
            ~~~~~~~

  node_modules/shaka-player/dist/shaka-player.compiled.d.ts:2316:7
    2316   let STOPPED : string ;
               ~~~~~~~
    'STOPPED' was also declared here.

node_modules/@types/chrome/chrome-cast/index.d.ts:122:16 - error TS2451: Cannot redeclare block-scoped variable 'isAvailable'.

122     export var isAvailable: boolean;
                   ~~~~~~~~~~~

  node_modules/shaka-player/dist/shaka-player.compiled.d.ts:2320:7
    2320   let isAvailable : boolean ;
               ~~~~~~~~~~~
    'isAvailable' was also declared here.

node_modules/@types/chrome/chrome-cast/index.d.ts:206:18 - error TS2300: Duplicate identifier 'ApiConfig'.

206     export class ApiConfig {
                     ~~~~~~~~~

  node_modules/shaka-player/dist/shaka-player.compiled.d.ts:2269:9
    2269   class ApiConfig {
                 ~~~~~~~~~
    'ApiConfig' was also declared here.

node_modules/@types/chrome/chrome-cast/index.d.ts:232:18 - error TS2300: Duplicate identifier 'Error'.

232     export class Error {
                     ~~~~~

  node_modules/shaka-player/dist/shaka-player.compiled.d.ts:2276:9
    2276   class Error {
                 ~~~~~
    'Error' was also declared here.

node_modules/@types/chrome/chrome-cast/index.d.ts:280:18 - error TS2300: Duplicate identifier 'SessionRequest'.

280     export class SessionRequest {
                     ~~~~~~~~~~~~~~

  node_modules/shaka-player/dist/shaka-player.compiled.d.ts:2309:9
    2309   class SessionRequest {
                 ~~~~~~~~~~~~~~
    'SessionRequest' was also declared here.

node_modules/@types/chrome/chrome-cast/index.d.ts:300:18 - error TS2300: Duplicate identifier 'Session'.

300     export class Session {
                     ~~~~~~~

  node_modules/shaka-player/dist/shaka-player.compiled.d.ts:2293:9
    2293   class Session {
                 ~~~~~~~
    'Session' was also declared here.

node_modules/@types/chrome/chrome-cast/index.d.ts:452:18 - error TS2300: Duplicate identifier 'Receiver'.

452     export class Receiver {
                     ~~~~~~~~

  node_modules/shaka-player/dist/shaka-player.compiled.d.ts:2286:9
    2286   class Receiver {
                 ~~~~~~~~
    'Receiver' was also declared here.
```

Not sure it's about interaction with other libraries. Even the simplest TS project doesn't compile.
```typescript
// index.ts
import { polyfill, Player } from 'shaka-player';
polyfill.registerAll();
const player = new Player();
player.load('https://dash.akamaized.net/akamai/bbb_30fps/bbb_30fps.mpd');
```
Compiling:
```bash
$ npm install shaka-player
[...]

$ tsc --lib dom,es2015 index.ts
index.ts:1:34 - error TS2306: File '/redacted/node_modules/shaka-player/dist/shaka-player.compiled.d.ts' is not a module.

1 import { polyfill, Player } from 'shaka-player';
                                   ~~~~~~~~~~~~~~

node_modules/shaka-player/dist/shaka-player.compiled.d.ts:489:62 - error TS2694: Namespace 'google.ima.dai.api' has no exported member 'StreamRequest'.

[...]

Found 10 errors.
```
Only workaround I found was to delete the `.d.ts` file...

--- a/build/build.py
+++ b/build/build.py
@@ -316,7 +316,7 @@ class Build(object):
       return False
 
     generated_externs = [extern_generator.output]
-    shaka_externs = shakaBuildHelpers.get_all_js_files('externs')
+    shaka_externs = shakaBuildHelpers.get_all_js_files('externs/shaka')
     if self.has_ui():
       shaka_externs += shakaBuildHelpers.get_all_js_files('ui/externs')
     ts_def_generator = compiler.TsDefGenerator(

